pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -798,6 +806,11 @@ func (nc *Controller) monitorNodeHealth() error {
 			name := node.Name
 			node, err = nc.kubeClient.CoreV1().Nodes().Get(context.TODO(), name, metav1.GetOptions{})
 			if err != nil {
+				// If the node no longer exists, delete the corresponding health data from nodeHealthMap.
+				if apierrors.IsNotFound(err) {
+					nc.nodeHealthMap.delete(name)
+					return true, nil
```

Comment:
```suggestion
```

Reply:
Sure, let me have a try.

Reply:
Well, I found it's hard to test the condition. This will only happen when the node has listed in #L770 and soon be deleted in cluster. After that #L805 will still add the node health into health map. It is not possible to test the `monitorNodeHealth` function because we need to delete it when #L770 has been run. @wzshiming

Reply:
By the way, I don't figure out why `return true, nil` should be removed. When the node has been deleted, there is no reason to try updating node health again.